fedimint-server/src/config.rs

```diff
@@ -143,37 +148,55 @@ pub struct ServerConfigParams {
 }
 
 impl ServerConfigConsensus {
-    pub fn to_client_config_try(
+    pub fn try_to_config_response(
         &self,
         module_config_gens: &ModuleInitRegistry,
-    ) -> anyhow::Result<ClientConfig> {
-        Ok(ClientConfig {
+    ) -> anyhow::Result<ConfigResponse> {
+        let modules: BTreeMap<ModuleInstanceId, ModuleConfigResponse> = self
+            .modules
+            .iter()
+            .map(|(module_instance_id, v)| {
+                let kind = v.kind();
+                Ok((
+                    *module_instance_id,
+                    module_config_gens
+                        .get(kind)
+                        .ok_or_else(|| format_err!("module config gen not found: {kind:?}"))?
+                        .to_config_response(v.value().clone())?,
+                ))
+            })
+            .collect::<anyhow::Result<_>>()?;
+
+        let mut engine = HashEngine::default();
```

Comment:
Ad-hoc hashing by listing each field seem very error prone. We should either hash the whole thing, or if not possible for some reason - have a method that all parts of the code can re-use.

Reply:
I agree, but I would need to modify the `Encodable` macro so we could ignore certain fields like the untyped module json and encode it using the typed version instead.  I'm not sure how to do that but maybe you could?

Reply:
We can implement `Encodable` manually if need be. Not sure if it helps here, but I think it might.

Reply:
Alternatively, we should convert the config to another version that is purely encodable before hashing. We only need to be able to do it one way anyway, I think?

We should totally `impl (En)Decodable for serde_json::Value`, BTW!

Reply:
> We should totally `impl (En)Decodable for serde_json::Value`, BTW!

Maybe sort term, long term I'd much rather make the type erased cfg a dyn newtype that can be en/decoded and serialized/deserialized. That would require another decoder fn, but that's ok imo.